client.go

```diff
@@ -179,3 +220,42 @@ func (c *StatsdClient) SendEvent(e event.Event) error {
 	}
 	return nil
 }
+
+func (c *StatsdClient) collector() {
+	for {
+		select {
+		case <- c.shutdown:
+			c.shutdown <- struct{}{}
+			return
+		case <- c.Timer.C:
+			c.statsMutex.Lock()
+			if c.flushStatBuffer(){
+				c.startTimer()
+			} else {
+				c.timerRunning = false
+			}
+			c.statsMutex.Unlock()
+		}
+	}
+}
+
+func (c *StatsdClient) flushStatBuffer() bool {
+	if c.buffer.Len() == 0 {
+		return false
+	}
+	_, err := c.conn.Write(c.buffer.Bytes())
+	if err == nil {
+		c.buffer.Reset()
```

Comment:
we would overflow buffer is we don't reset it? does it make sense to ignore errors here and reset always?